### PR TITLE
Fix TypeScript compilation performance (closes #1591)

### LINKED ISSUE #1591: TypeScript Performance
### Are you requesting a feature or reporting a bug?

Bug / Perf Issue

### What is the current behavior?

Using the OOTB TypeScript support causes *significant* performance overhead.

The following are results from precompiling w/ tsc and running testcafe on the output vs. running testcafe on TypeScript files directly:

```
> npm run benchmark

> testcafe-slow-modules-example@ benchmark /Users/caseywebb/Code/testcafe-example
> echo TestCafé Version: $(testcafe --version) && npm run benchmark:prebuild && npm run benchmark:native

TestCafé Version: 0.16.2

> testcafe-slow-modules-example@ benchmark:prebuild /Users/caseywebb/Code/testcafe-example
> time sh -c 'tsc && testcafe chrome fixture/dist/index.js' > /dev/null


real    0m7.073s
user    0m3.328s
sys     0m0.370s

> testcafe-slow-modules-example@ benchmark:native /Users/caseywebb/Code/testcafe-example
> time testcafe chrome fixture/src/index.ts > /dev/null


real    0m20.673s
user    0m16.759s
sys     0m0.997s
```

### What is the expected behavior?

Compilation should be comparable with precompiling tests with `tsc`

### How would you reproduce the current behavior (if this is a bug)?

[funny you should ask](https://github.com/caseyWebb/testcafe-slow-typescript-example)

### System Details

* operating system: macOS 10.12 and Windows 10
* testcafe version: 0.16.2
* node.js version: 8.1.3

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @AlexanderMoskovkin 
Tested on https://github.com/caseyWebb/testcafe-slow-typescript-example. I'm going to run it on Travis CI for demonstration, and will post a link after it will be done.

### CODE REVIEW

File: src/compiler/test-file/formats/typescript.js [hunk 1]
```diff
@@ -36,29 +41,46 @@ export default class TypeScriptTestFileCompiler extends APIBasedTestFileCompiler
...
             TypeScriptTestFileCompiler._reportErrors(diagnostics);
 
-        var result = ts.transpileModule(code, {
-            compilerOptions:     opts,
-            fileName:            filename,
-            renamedDependencies: { testcafe: APIBasedTestFileCompilerBase.EXPORTABLE_LIB_PATH }
-        });
+        // NOTE: First argument of emit() is a source file for compilation. If it's undefined, all files in <program> will be compiled.
+        // <program> contains the file specified in createProgram() plus all its dependencies. This mode is a lot faster than
+        // compiling files one-by-one, and used in tsc CLI compiler.
```
Comment (VasilyStrelyaev) [inline src/compiler/test-file/formats/typescript.js]: what is `<program>`?

**Comment by AlexanderMoskovkin** [comment]
FPR
/cc @churkin @helen-dikareva 

It's also checked on tests of our ASP team. 
They have tests with a lot of helpers. Without the fix it was about 35s from run tests command  to browser opening if one test is chosen. Now it's about 5s,

The next stage of improving will be finding a way to compile faster a pack of tests (now we compile each test separately so it takes much time).

File: src/compiler/test-file/formats/typescript.js (hunk 1 above)
Comment (AlexanderMoskovkin) [inline src/compiler/test-file/formats/typescript.js]: it's an object created by the `ts.createProgram([filename], opts);` execution (see above)

Comment (VasilyStrelyaev) [inline src/compiler/test-file/formats/typescript.js]:
```
        // NOTE: First argument of emit() is a source file to be compiled. If it's undefined, all files in <program> will be compiled.
        // <program> contains a file specified in createProgram() plus all its dependencies. This mode is much faster than
        // compiling files one-by-one, and it is used in the tsc CLI compiler.
```

**Comment by AlexanderMoskovkin** [comment]
FPR

**Pull request merged**